tools/jdk/local_java_repository.bzl

```diff
@@ -12,13 +12,25 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Java repository implementation.
+"""Rules for importing and registering a local JDK."""
 
-Creates a local repository using jdk.BUILD file.
+def _detect_java_version(repository_ctx, java_bin):
+    properties_out = repository_ctx.execute([java_bin, "-XshowSettings:properties"]).stderr
 
-When Java executable is not present it creates a BUILD file with target "jdk"
-displaying an error message.
-"""
+    version_property = "java.version = "
+    versions = [
+        property.lstrip()[len(version_property):].rstrip()
```

Comment:
This is taking something of the form "java.version = 11.16.7" and parsing it to just "11.16.7", right? The lstrip and rstrip are a bit confusing, as is the multi-line list comprehension. I'd consider re-writing as a standard for loop (possibly in a separate function, but that might be overkill). But then you can just use `strip()` instead of `lstrip()` and `rstrip()`.

Reply:
I didn't want to use `strip`, because it also removes inner spaces (https://docs.bazel.build/versions/3.7.0/skylark/lib/string.html#strip), but this is actually fine. Will re-write as a loop and use `strip`.

Reply:
Actually for loop didn't look that nice - jumping out on first occurrence and handling case when version is not found looks worse. I opted for a more readable functional implementation.